frontend/evaluations/views.py

```diff
@@ -3,114 +3,116 @@
 from django.middleware.csrf import get_token
 from django.shortcuts import render, redirect
 from .models import Evaluation_block, Evaluation
+import re, unicodedata
 import requests
+import queue
+
+# input_text_regex = re.compile(r'^[\x20-\x7E\p{L}\p{N}]+$')
+request_queue = queue.Queue()
+
+def create_evaluation_fnc(evaluation_block, evaluation_dict):
+    new_evaluation = Evaluation.objects.create(
+        evaluation_block=evaluation_block,
+
+        claim=evaluation_dict.get("claim"),
+        label=evaluation_dict.get("label"), 
+        supports=evaluation_dict.get("supports"), 
+        refutes=evaluation_dict.get("refutes"),
+        ei=evaluation_dict.get("ei"),
+        nei=evaluation_dict.get("nei"),
+        evidence=evaluation_dict.get("evidence"),
+        justify=evaluation_dict.get("justify"),
+        url=evaluation_dict.get("url")
+    )
+    evaluation_dict["id"] = new_evaluation.id # ! Adding id to the obtained JSON -> passing to feedbacks app
+    evaluation_dict["evaluation_block"] = evaluation_block.id
+
+
+def eval_fnc(request, input_text, backend_url):
+
+    # # * Check for invalid characters
+    # if not input_text_regex.match(input_text):
+    #     return JsonResponse({"error_msg": "Invalid input. The input contains invalid characters.", "status": 400})    
+
+    # * Check for authentication
+    user = request.user
+    if user.is_authenticated:
+        validated_text = requests.get(backend_url, params={"text" : input_text})
+
+        if validated_text.status_code == 503:
+            request_queue.put((request, input_text, backend_url))
+            return JsonResponse({"status" : 503})
+        
+        while not request_queue.empty():
```

Comment:
I realized that it is not a bad thing to make an interval that waits before asking. So you don't keep calling my endpoint.